lib/extract/netcdf/lfric/kernel_data_netcdf.jinja

```diff
@@ -0,0 +1,249 @@
+            DeclareFieldDouble, &
+            DeclareFieldVectorDouble, &
+            {{all_declares|join(", &\n"+indent) }}
+
+        !> The generic interface for providing the value of variables,
+        !! which in case of the read-only verification either computes
+        !! the checksum (before a kernel), or compares a checksum (after
+        !! a kernel call).
+        generic, public :: ProvideVariable => &
+            WriteFieldDouble,       &
+            WriteFieldVectorDouble, &
+            {{all_writes|join(", &\n"+indent) }}
+
+        !> The generic interface for reading in variables previously
+        !! written. This is not part of the standard PSyData API, but
+        !! it is convenient for a driver that e.g. reads previously written
+        !! files. The driver does not use LFRic fields (only Fortran arrays),
+        !! so there is no need to provided field-specific read functions.
+        generic, public :: ReadVariable => &
+            {{all_reads|join(", &\n"+indent) }}
+
+    end type extract_PSyDataType
+
+Contains
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine declares a double precision LFRic field.
+    !! It uses the 1d-double precision array function provided by the base class.
+    !! @param[inout] this The instance of the extract_PsyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[in] value The value of the variable.
+    subroutine DeclareFieldDouble(this, name, value)
+        use field_mod, only : field_type, field_proxy_type
+        implicit none
+
+        class(extract_PsyDataType), intent(inout), target :: this
+        character(*), intent(in)                          :: name
+        type(field_type), intent(in)                      :: value
+
+        type(field_proxy_type) :: value_proxy
+
+        value_proxy = value%get_proxy()
+        call this%ExtractNetcdfBaseType%DeclareArray1dDouble(name, value_proxy%data)
+    end subroutine DeclareFieldDouble
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine writes the value of a double precision LFRic field
+    !! to the NetCDF file. It uses the 1d-double precision array
+    !! writing function from the base class.
+    !! @param[inout] this The instance of the extract_PsyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[in] value The value of the variable.
+    subroutine WriteFieldDouble(this, name, value)
+        use field_mod, only: field_type, field_proxy_type
+        implicit none
+
+        class(extract_PsyDataType), intent(inout), target :: this
+        character(*), intent(in)                          :: name
+        type(field_type), intent(in)                      :: value
+
+        type(field_proxy_type) :: value_proxy
+
+        value_proxy = value%get_proxy()
+        call this%ExtractNetcdfBaseType%WriteArray1dDouble(name, value_proxy%data)
```

Comment:
In future, could we have a `WriteArray1d` Fortran interface and then let the compiler worry about which actual implementation to call?

Reply:
We already have a generic `ProvideVariable` interface, which matches to `WriteArray1dDouble` (or ``WriteArray1dReal`` or so). I just had to use it :grin:

Reply:
Aha!